RecoLocalTracker/SiPixelRecHits/plugins/SiPixelRecHitConverter.cc

```diff
@@ -185,9 +189,34 @@ namespace cms {
 
     const edmNew::DetSetVector<SiPixelCluster>& input = *inputhandle;
 
-    edmNew::DetSetVector<SiPixelCluster>::const_iterator DSViter = input.begin();
+    // fill cluster arrays
+    auto hmsp = std::make_unique<uint32_t[]>(gpuClustering::MaxNumModules + 1);
+    auto hitsModuleStart = hmsp.get();
+    std::array<uint32_t, gpuClustering::MaxNumModules + 1> clusInModule{};
+    for (auto DSViter = input.begin(); DSViter != input.end(); DSViter++) {
+      unsigned int detid = DSViter->detId();
+      DetId detIdObject(detid);
+      const GeomDetUnit* genericDet = geom.idToDetUnit(detIdObject);
+      auto gind = genericDet->index();
+      // FIXME to be changed to support Phase2
+      if (gind >= int(gpuClustering::MaxNumModules))
+        continue;
+      auto const nclus = DSViter->size();
+      assert(nclus > 0);
+      clusInModule[gind] = nclus;
+      numberOfClusters += nclus;
+    }
+    hitsModuleStart[0] = 0;
+    assert(clusInModule.size() > gpuClustering::MaxNumModules);
+    for (int i = 1, n = clusInModule.size(); i < n; ++i)
+      hitsModuleStart[i] = hitsModuleStart[i - 1] + clusInModule[i - 1];
+    assert(numberOfClusters == int(hitsModuleStart[gpuClustering::MaxNumModules]));
+
+    // yes a unique ptr of a unique ptr so edm is happy and the pointer stay still...
+    iEvent.emplace(tHost_, std::move(hmsp));  // hmsp is gone, hitsModuleStart still alive and kicking...
```

Comment:
@VinInn I reorganised this a little to fix the merge conflicts and make it slightly more in line with the upstream code.
Could you confirm that this still does what it is supposed to do ?

Reply:
as I said in all my recent talks to DPG this is a huge pain&mess until not merged in official CMSSW...
I will have a look, BUT we really need to consolidate this code with the other 3 producers

Reply:
at first glance looks ok. would need to check the diff generated by the merge conflicts...